name: Dependencies Review & Audit

on:
  # "This run was triggered by the "push" event, which is unsupported.
  #  Please ensure you are using the "pull_request" event for this workflow."
  pull_request:
  workflow_call: # SRC: https://docs.github.com/en/actions/using-workflows/reusing-workflows
    inputs:
      ssh-private-key:
        type: string
        description: SSH private key
        required: false

      yarn_audit_cli_options:
        type: string
        description: Options to pass to the yarn
        required: false
        default: --groups dependencies

# TODO
# concurrency:
#   group: ${{ github.ref }}-${{ github.workflow }}
#   cancel-in-progress: true

permissions:
  contents: read

jobs:
  dependency-review:
    uses: percebus/github-actions-common/.github/workflows/dependency-review.yml@main
    with:
      ssh-private-key: ${{ inputs.ssh-private-key }}

  audit:
    uses: ./.github/workflows/audit.yml
    with:
      ssh-private-key: ${{ inputs.ssh-private-key }}
      yarn_audit_cli_options: ${{ inputs.yarn_audit_cli_options }}
